client/state/marketing/selectors.ts

```diff
@@ -35,3 +40,23 @@ export function getFullJetpackSaleCouponDiscountRatio( state: AppState ): number
 	const discount = getJetpackSaleCoupon( state )?.final_discount || 0;
 	return discount / 100;
 }
+
+interface DiscountMap {
+	[ productSlug: string ]: number;
+}
+
+export function getJetpackCouponDiscountMap( state: AppState ): DiscountMap {
+	const discount = ( 100 - ( getJetpackSaleCoupon( state )?.discount || 0 ) ) / 100;
+	const discountMap: DiscountMap = {};
+
+	[ JETPACK_RESET_PLANS_BY_TERM, JETPACK_PRODUCTS_BY_TERM ].forEach( ( productSlugMap ) =>
```

Comment:
Minor, but I think you could have just one `forEach` by merging these arrays.
```suggestion
	[ ...JETPACK_RESET_PLANS_BY_TERM, ...JETPACK_PRODUCTS_BY_TERM ].forEach( ( { yearly } ) =>
```